homework/hw7/task2_backspace.py

```diff
@@ -27,50 +28,42 @@ def calc_backspace(text: str) -> str:
     return "".join(new_string)
 
 
-"""Не знаю, какая лучше"""
-
-
-def c(text: str) -> str:
-    """Re-write text with backspace instead of #."""
-    new_string = ""
-    for char in text:
-        if char != "#":
-            new_string += char
-        elif len(new_string) > 0:
-            new_string = new_string[:-1]
-    return new_string
-
-
 def backspace_compare(first: str, second: str) -> bool:
     """Compare if two given strings are equal."""
-    return c(first) == c(second)
+    return calc_backspace(first) == calc_backspace(second)
 
 
 """some proof of concept"""
 
 
 def random_fun(string: str, string2: str) -> bool:  # noqa: CCR001
     """Compare if two given strings are equal."""
```

Comment:
Not sure if there anything else to simplify.

Reply:
It's ok to use a separated function to split complex solution